src/inmanta/env.py

```diff
@@ -1237,11 +1237,47 @@ def install_from_list(
         upgrade: bool = False,
         upgrade_strategy: PipUpgradeStrategy = PipUpgradeStrategy.ONLY_IF_NEEDED,
     ) -> None:
-        if not self.__using_venv:
+        if not self._using_venv:
             raise Exception(f"Not using venv {self.env_path}. use_virtual_env() should be called first.")
         super(VirtualEnv, self).install_from_list(requirements_list, upgrade=upgrade, upgrade_strategy=upgrade_strategy)
 
 
+class ReentrantVirtualEnv(VirtualEnv):
+    """
+    A virtual env that can be de-activated and re-activated
+
+    This allows faster reloading due to improved caching of the working set
+
+    This is intended for use in testcases to require a lot of venv switching
+    """
+
+    def __init__(self, env_path: str) -> None:
+        super(ReentrantVirtualEnv, self).__init__(env_path)
+        self.working_set = None
+
+    def deactivate(self):
+        self._using_venv = False
+        self.working_set = pkg_resources.working_set
```

Comment:
This is tricky. I don't think this implementation suffices in general. Then again, all other steps are done in the snippetcompiler and cleanup fixtures already. But if we're putting this in `src` that might not suffice.
I wonder if it wouldn't be simpler to just hack this in the snippetcompiler directly rather than to have this class. I think it would only require two lines, wouldn't it?

Reply:
I can put this sub-class in the test folder ?

Reply:
That would be a lot better already. Then we can just say that other cleanup happens in the autouse fixture or something like that.